storage/innobase/dict/dict0load.cc

```diff
@@ -1800,6 +1852,208 @@ dict_load_virtual_low(
 	return(NULL);
 }
 
+/** Error message for a delete-marked record in dict_load_column_added_low() */
+static const char* dict_load_column_added_del = "delete-marked record in SYS_COLUMNS_ADDED";
+
+/** Load a table added column definition from a SYS_COLUMNS_ADDED record to dict_table_t.
+@return	error message
+@retval	NULL on success */
+static
+const char*
+dict_load_column_added_low(
+	dict_table_t*	table,		/*!< in/out: table, could be NULL */
+	mem_heap_t*		heap,		/*!< in/out: memory heap for temporary storage */
+	const rec_t*	rec,		/*!< in: SYS_COLUMNS_ADDED record */
+	dict_index_t*   sys_index,
+	table_id_t*		table_id_out,
+	ulint*			pos_out,
+	char**			def_val_out,
+	ulint*			def_val_len_out)
+{
+	const byte*		field;
+	const byte*		def_val = NULL;
+	ulint			def_val_len = 0;
+	table_id_t		table_id;
+	ulint			len;
+	ulint			pos;
+
+	if (rec_get_deleted_flag(rec, 0)) {
+		return(dict_load_column_added_del);
+	}
+
+	if (rec_get_n_fields_old(rec) != DICT_NUM_FIELDS__SYS_COLUMNS_ADDED) {
+		return("wrong number of columns in SYS_COLUMNS_ADDED record");
+	}
+
+	field = rec_get_nth_field_old(
+		rec, DICT_FLD__SYS_COLUMNS_ADDED__TABLE_ID, &len);
+	if (len != 8) {
+err_len:
+		return("incorrect column length in SYS_COLUMNS_ADDED");
+	}
+
+	table_id = mach_read_from_8(field);
+	if (table && table->id != mach_read_from_8(field)) {
+		return("SYS_COLUMNS_ADDED.TABLE_ID mismatch");
+	}
+
+	field = rec_get_nth_field_old(
+		rec, DICT_FLD__SYS_COLUMNS_ADDED__POS, &len);
+	if (len != 4) {
+		goto err_len;
+	}
+
+	pos = mach_read_from_4(field);
+
+	rec_get_nth_field_offs_old(
+		rec, DICT_FLD__SYS_COLUMNS_ADDED__DB_TRX_ID, &len);
+	if (len != DATA_TRX_ID_LEN && len != UNIV_SQL_NULL) {
+		goto err_len;
+	}
+	rec_get_nth_field_offs_old(
+		rec, DICT_FLD__SYS_COLUMNS_ADDED__DB_ROLL_PTR, &len);
+	if (len != DATA_ROLL_PTR_LEN && len != UNIV_SQL_NULL) {
+		goto err_len;
+	}
+
+	if (UNIV_UNLIKELY(rec_offs_nth_extern_old(rec, 
+		DICT_FLD__SYS_COLUMNS_ADDED__DEFAULT_VALUE))) {
+		ulint       offsets_[REC_OFFS_NORMAL_SIZE];
+		ulint *     offsets = offsets_;
+
+		ut_a(rec);
+		rec_offs_init(offsets_);
+
+		offsets = rec_get_offsets(rec,sys_index,offsets,ULINT_UNDEFINED,&heap);
+
+		def_val = btr_rec_copy_externally_stored_field(
+				rec, offsets, 
+				dict_table_page_size(table),
+				DICT_FLD__SYS_COLUMNS_ADDED__DEFAULT_VALUE, &def_val_len, heap);
```

Comment:
MariaDB starting with 10.2.1
BLOB and TEXT columns can now have a DEFAULT value.

Because of it, the default value of blob columns may be very large.